pkg/gateway/client/identity.go

```diff
@@ -220,6 +220,12 @@ func (c *Client) ensureIdentity(ctx context.Context, tx *gorm.DB, id *types.Iden
 				userChanged = true
 			}
 
+			if user.Username != id.ProviderUsername {
+				user.Username = id.ProviderUsername
+				user.HashedUsername = hash.String(user.Username)
+				userChanged = true
+			}
```

Comment:
Curious about this change. The original intention was that a user should be able to change their username and we just use the username from the provider to start with.

What's the intention now?

Reply:
If user changed the auth provider to login, the identify will also change but the new username is not reflected on the user. For exmaple, when user login with google, they got assigned with google username which is a random string. then later they login with github(with the same email account), when we update existing user to reflect the latest identity, we will need to update google's username to github's. This is from my understanding that username should get changed when user changed their auth provider(in this case, when email is the same and we merge accounts).